src/wallet/db.cpp

```diff
@@ -408,11 +408,7 @@ bool BerkeleyBatch::VerifyEnvironment(const fs::path& file_path, std::string& er
     LogPrintf("Using wallet %s\n", file_path.string());
 
     // Wallet file must be a plain filename without a directory
-    if (walletFile != fs::basename(walletFile) + fs::extension(walletFile))
-    {
-        errorStr = strprintf(_("Wallet %s resides outside wallet directory %s"), walletFile, walletDir.string());
-        return false;
-    }
+    assert(walletFile == fs::path(walletFile).filename().string());
```

Comment:
Suggestion, either drop the assertion (`GetWalletEnv` always returns a plain filename) or make it complete because:
```cpp
std::cout << path("/").filename();            // outputs "/"
std::cout << path(".").filename();            // outputs "."
std::cout << path("..").filename();           // outputs ".."
```
so for these cases the assertion passes.